### PR TITLE
Catalog sync service

### PR DESCRIPTION
**Pull request by sthaha**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/design/type.go [hunk 1]
```diff
@@ -299,3 +300,10 @@ var HubService = Type("HubService", func() {
 
 	Required("name", "status")
 })
+
+var Job = ResultType("application/vnd.hub.job", "Job", func() {
+	Description("The Job type describes a ")
```
Comment (SM43) [inline api/design/type.go:305]: Description :D

Comment (sthaha) [reply api/design/type.go]: 😝

File: api/design/catalog.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			Required("token")
+		})
+		Result(Job)
+
+		HTTP(func() {
+			POST("/catalog/refresh")
+			Header("token:Authorization")
+
+			Response(StatusOK)
+			Response("internal-error", StatusInternalServerError)
```
Comment (SM43) [inline api/design/catalog.go:43]: `Response("not-found", StatusNotFound)`

File: api/go.mod [hunk 3]
```diff
@@ -3,25 +3,74 @@ module github.com/tektoncd/hub/api
 go 1.14
 
 require (
+	cloud.google.com/go v0.57.0 // indirect
```
Comment (SM43) [inline api/go.mod:6]: `go mod tidy` :smile:

File: api/pkg/parser/catalog.go [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+	for _, d := range dirs {
+		if d.IsDir() {
+			count++
+		}
+	}
+	return count
+}
+
+func (c CatalogParser) parseResource(kind, kindPath string, f os.FileInfo) (*Resource, Result) {
+	// TODO(sthaha): move this to a different package that can scan a CatalogParser
```
Comment (SM43) [inline api/pkg/parser/catalog.go:147]: wanna move this now or later?

File: api/pkg/service/catalog/catalog_http_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+		server.MountRefreshHandler,
+		catalog.NewRefreshEndpoint(NewServiceTest(tc), service.JWTAuth))
+	return checker
+}
+
+func TestRefresh_Http(t *testing.T) {
+	tc := testutils.Setup(t)
+	testutils.LoadFixtures(t, tc.FixturePath())
+
+	data := []byte(`{"name": "catalog-official","org":"tektoncd"}`)
```
Comment (SM43) [inline api/pkg/service/catalog/catalog_http_test.go:49]: we can remove this now

**Pull request merged**